### PR TITLE
handle scraped locale

### PR DESCRIPTION
**Pull request by TonySchroeder**
close https://github.com/progwise/george-ai/issues/65

- All currently supported languages in Strapi are fetched.
- From those, a filter is applied to identify languages supported in prompts.
- If there is no match, the process is terminated.
- A check is performed to see if scrapedLanguage is included; if not, the page is skipped.
- Some type adjustments are made.

### CODE REVIEW

**Comment by moncapitaine** [comment]
Is the page skipped completly or are the links still analyzed for other pages?

File: apps/playwright-scraper/src/locales.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { graphql } from './gql'
+import { strapiClient } from './strapi-client'
+
+export const getAllStrapiLocales = async () => {
```
Comment (moncapitaine) [inline apps/playwright-scraper/src/locales.ts:4]:
```suggestion
export const getStrapiLocales = async () => {
```

File: apps/playwright-scraper/src/main.ts [hunk 2]
```diff
@@ -16,18 +18,51 @@ const processPage = async (url: string): Promise<void> => {
   const browser = await playwright['chromium'].launch({ headless: true })
   const context = await browser.newContext()
 
+  let runCounter = 0 // Counter
+
   const urlsDone: Array<string> = []
   let urlsTodo = [url]
+  const strapiLocales = await getAllStrapiLocales()
+  // eslint-disable-next-line unicorn/no-array-callback-reference
```
Comment (moncapitaine) [inline apps/playwright-scraper/src/main.ts:26]: do a safe call and do not exclude the unicorn here.

File: apps/playwright-scraper/src/main.ts [hunk 3]
```diff
@@ -16,18 +18,51 @@ const processPage = async (url: string): Promise<void> => {
...
+      urlsTodo = [
+        ...new Set(
+          [...urlsTodo, ...scrapeResult.links].filter(
+            (url) => !urlsDone.includes(url),
+          ),
+        ),
+      ]
+
+      if (
+        // eslint-disable-next-line unicorn/prefer-includes
```
Comment (moncapitaine) [inline apps/playwright-scraper/src/main.ts:52]: Why is this needed?

File: apps/playwright-scraper/src/prompts.ts [hunk 4]
```diff
@@ -24,3 +24,8 @@ export const prompts = {
     ],
   },
 }
+
+export type Language = keyof typeof prompts
+
+export const isLanguage = (language: string): language is Language =>
```
Comment (moncapitaine) [inline apps/playwright-scraper/src/prompts.ts:30]: is the '''is''' operator really used here?

File: apps/playwright-scraper/src/main.ts (hunk 2 above)
Comment (TonySchroeder) [reply apps/playwright-scraper/src/main.ts:26]: const promptsLocales = strapiLocales.filter((locale) => isLanguage(locale));

in that case the array would not have the correct type and that is what the function is for

File: apps/playwright-scraper/src/main.ts (hunk 3 above)
Comment (TonySchroeder) [reply apps/playwright-scraper/src/main.ts:52]: unicorn wants includes to be used, but with this the types do not match.

File: apps/playwright-scraper/src/prompts.ts (hunk 4 above)
Comment (TonySchroeder) [reply apps/playwright-scraper/src/prompts.ts:30]: Yes, the is operator confirms that the given language string is actually a key in the prompts object, ensuring type safety.

File: apps/playwright-scraper/src/main.ts (hunk 3 above)
Comment (TonySchroeder) [reply apps/playwright-scraper/src/main.ts]: delete eslint-disable-next-line

**Review by moncapitaine — APPROVED** [review]
